mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Functions for multivariate richardson extrapolation as defined in
+:cite:`Russo_2024_LRE`.
+"""
+
+import warnings
+from itertools import product
+from typing import Any, Optional
+
+import numpy as np
+from cirq import Circuit
+from numpy.typing import NDArray
+
+from mitiq.lre.multivariate_scaling.layerwise_folding import (
+    _get_scale_factor_vectors,
+)
+
+
+def _full_monomial_basis_term_exponents(
+    num_layers: int, degree: int
+) -> list[tuple[int, ...]]:
+    """Finds exponents of monomial terms required to create the sample matrix.
+
+    $Mj(λ_i, d)$ is the basis of monomial terms for $l$-layers in the input
```

Comment:
If you're going to reference a piece of notation make sure to provide a reference for where it comes from. There is a reference in the top of this file, but maybe an equation number would be helpful? WDYT?